### DISCUSSION TITLE
Shiny App menggunakan data dari MySQL

### DISCUSSION THREAD

**Discussion by vidric** [teamalgoritma/community#578] [Q&A] [answered]
Sy telah membuat dashboard dengan sumber data dari mysql (online).
berikut code di global.R :
`con <- dbConnect(drv = MySQL(), 
                 host = "5.181.216.212", 
                 dbname = "u1737929_spl_2", 
                 user = "******", 
                 password = "*********")

### get raw data ##########
data_battle_raw <- con %>%
  tbl("battle") %>%
  collect()

data_player_raw <- con %>%
  tbl("player") %>%
  collect()

data_card_raw <- con %>%
  tbl("card") %>%
  collect()

data_relation_battle_card_raw <- con %>%
  tbl("r_battle_card") %>%
  collect()`

Dashboard akan menampilkan error sebagai berikut:
**Unable to connect to worker after 60.00 seconds; startup took too long. Contact the author for more information.**

Sy ada 2 pertanyaan:
1. Bagaimana shiny app ini dapat mengelolah data yg besar, yg sumber datanya dari mysql?
2. Bagaiman shiny app ini dapat mengambil data baru dari mysql tanpa harus deploy ulang?

**Comment by idoali** [comment, marked as answer]
Terima kasih atas pertanyaannya pak

Shiny apps bisa mengolah data berukuran besar yang berasal dari Relational Database pak. Shiny apps melakukan pengolahan data ini dengan memanfaatkan library `DBI`. Library ini digunakan untuk membuat koneksi antara program R dengan database, termasuk yang ada di cloud. Setelah itu, library `DBI` juga memiliki fungsi untuk melakukan query, yang langkah-langkahnya sama seperti melakukan query seperti biasa. 

Untuk overview dan panduan singkatnya sendiri bisa dibaca disini pak. https://shiny.rstudio.com/articles/overview.html 

Pada artikel itu, script R dihubungkan dengan Relational Database dari AWS.
